app/src/main/java/co/tinode/tindroid/UiUtils.java

```diff
@@ -501,6 +514,43 @@ static String millisToTime(int millis) {
         return sb.append(sec).toString();
     }
 
+    // Builds and returns a video call message text given call direction, state and duration.
+    @NonNull
+    static Spanned videoCallMsg(boolean isOutgoing, String callState, int millis) {
+        String dir = isOutgoing ? "↗" : "↙";
+        Spannable cont = new SpannableString(dir);
+        cont.setSpan(new ForegroundColorSpan(
+                        "disconnected".equals(callState) ? Color.RED : Color.GREEN),
+                0, dir.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        cont.setSpan(new StyleSpan(Typeface.BOLD), 0, dir.length(), Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
+
+        StringBuilder sb = new StringBuilder();
+        sb.append(isOutgoing ? "Outgoing Call" : "Incoming Call");
```

Comment:
If these strings are user-visible, then they should be saved as string resources.